hotsos/client.py

```diff
@@ -408,6 +429,15 @@ def run(self):
                     content = self._run(plugin)
                     if content:
                         self.summary.update(plugin, content.get(plugin))
+        except Exception as e:
+            print('\nException running plugins:', e)
+            if temp_log:
+                print('See temp log file for possible details:', temp_log)
+            raise
         finally:
             log.name = 'hotsos.client'
+            self.clear_plugin_env()
             self.teardown_global_env()
+
+        # This will only get called if no exceptions
+        self.remove_temp_log_file()
```

Comment:
this can go inside the ```try:``` clause at the end since we are executing it if no exception occured

Reply:
Done